### ISSUE TITLE
Scope to improve command submission.

### ISSUE THREAD

**Issue by Sarbojit2019** [CHIP-SPV/chipStar#224]
I was looking into hipMemcpy implementation in Chip_spv, there are many extra commands getting submitted which if skipped will get much better execution time. I wrote tiny reproducer in order to support my view, portion of reproduce is as below

```
int main() {
......
  for (int i=0; i < iter; ++i) {
    printf("Memcpy starts ------------------------------------>\n");
    hipMemcpy(dst, hst, size, hipMemcpyHostToDevice);
    printf("Memcpy ends -------------------------------------->\n");
  }  
.....
}
```
![image](https://user-images.githubusercontent.com/52556743/198268911-42e193b7-8c3d-4c33-8650-4f82b37e59c2.png)
Command submitted in the blue circle is unnecessary and can be completely skipped. Also I am not sure why is extra barrier needed in case of sync copy submission which is happening in the red circle. Since this is sync call, there is no way any other command gets submitted to the queue as host thread will be blocked until copy is completed.

**Comment by pjaaskel** [comment]
Synchronizing after each command is a solid approach but doesn't utilize asynch execution. Something to optimize for 1.1/2.0?

**Comment by pvelesko** [comment]
Fixed in #817
